ffi/capi/src/zoned_formatter.rs

```diff
@@ -57,20 +61,25 @@ pub mod ffi {
             )))
         }
 
-        /// Formats a [`IsoDateTime`] and [`CustomTimeZone`] to a string.
+        #[cfg(feature = "compiled_data")]
+        /// Formats a [`IsoDateTime`] and [`TimeZone`] to a string.
         pub fn format_iso_datetime_with_custom_time_zone(
             &self,
             datetime: &IsoDateTime,
-            time_zone: &CustomTimeZone,
+            time_zone: &TimeZone,
+            timezone_calculator: &TimeZoneCalculator,
```

Comment:
Nit: Do the same abstraction in FFI as you have in Rust, instead of passing the timezone calculator in here

Reply:
OK, I thought we were mostly finalized, but from https://github.com/unicode-org/icu4x/issues/5533 now I understand that we're not